shell/components/formatter/__tests__/LiveDate.test.ts

```diff
@@ -0,0 +1,82 @@
+import { mount } from '@vue/test-utils';
+import LiveDate from '@shell/components/formatter/LiveDate.vue';
+import dayjs from 'dayjs';
+
+const defaultMock = {
+  $store: {
+    getters: {
+      'prefs/get':   jest.fn(),
+      'i18n/t':      ( unitKey: string, { count }: any) => unitKey.split('.')[1] + (count > 1 ? 's' : ''),
+      'i18n/exists': () => false,
+
+    }
+  },
+};
+
+describe('component: LiveDate', () => {
+  it('should show a dash if no date is provided', () => {
+    const wrapper = mount(LiveDate, { mocks: defaultMock });
+    const element = wrapper.find('span');
+
+    expect(element.text()).toBe('-');
+  });
+
+  it('should say just now when the time provided is the current time', async() => {
+    const wrapper = await mount(LiveDate, {
+      mocks:     defaultMock,
+      propsData: { value: dayjs().toString() }
+    });
+
+    const element = wrapper.find('span');
+
+    expect(element.text()).toContain('Just now');
+  });
+
+  it('should display difference between value and current time', async() => {
+    const wrapper = await mount(LiveDate, { mocks: defaultMock, propsData: { value: Date.now() - 5000 } });
+
+    const element = wrapper.find('span');
+
+    expect(element.text()).toContain('5 secs');
+  });
+
+  // this test calls a component method directly: we're testing this because it's how 'live' formatters are used in sortabletable
+  it('should recompute and update the displayed value when prompted', async() => {
+    const wrapper = await mount(LiveDate, {
+      mocks:     defaultMock,
+      propsData: { value: dayjs().toString() }
+    });
+
+    let element = wrapper.find('span');
+
+    expect(element.text()).toContain('Just now');
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    await wrapper.vm.liveUpdate(Date.now());
```

Comment:
This is generating a linting error during build on e2e tests "Prepare Build" step. In order to fix this (it ain't pretty...) One idea is what I used (with Giuseppe's help) for my unit tests PR:
https://github.com/rancher/dashboard/pull/7420/files#diff-3768d1217b02196ee43bf62bee0f4965eeb8265fdea94db6a5cee1949df6d7ba

import these:
```
import { ExtendedVue, Vue } from 'vue/types/vue';
import { DefaultProps } from 'vue/types/options';
```

and instead of:
```
const wrapper = await mount(LiveDate, {
```

you do:
```
const wrapper = await mount(LiveDate as unknown as ExtendedVue<Vue, {}, {}, {}, DefaultProps>, {
```

TS fun here... 😛  That should fix it

Reply:
Oof, thank you! Who knows how long that would have taken me to work out myself D: